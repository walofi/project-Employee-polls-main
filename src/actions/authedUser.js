export const SET_AUTHED_USER = "SET_AUTHED_USER";
export const CHECK_AUTHED_USER = "CHECK_AUTHED_USER";

export function setAuthedUser(id) {
  return {
    type: SET_AUTHED_USER,
    id,
  };
}
