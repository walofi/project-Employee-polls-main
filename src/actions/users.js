export const RECEIVE_USERS = "RECEIVE_USERS";
export const ADD_ANSWER = "ADD_ANSWER";
export const ADD_QUESTION_TO_USER = "ADD_QUESTION_TO_USER";

export function receiveUsers(users) {
  return {
    type: RECEIVE_USERS,
    users,
  };
}

export function addAnswer({ authedUser, qid, answer }) {
  return {
    type: ADD_ANSWER,
    authedUser,
    qid,
    answer,
  };
}
export function addQuestionToUser({ authedUser, qid }) {
  return {
    type: ADD_QUESTION_TO_USER,
    authedUser,
    qid,
  };
}
