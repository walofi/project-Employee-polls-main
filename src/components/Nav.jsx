/* eslint-disable react/prop-types */
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import Image from "react-bootstrap/Image";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { setAuthedUser } from "../actions/authedUser";

const Navigation = (props) => {
  const navigate = useNavigate();
  const isActive = (path) => window.location.pathname === path;

  return (
    <Navbar className="container" fixed="top" bg="dark" variant="dark">
      <Navbar.Brand
        style={{
          cursor: "pointer",
        }}
        onClick={() => navigate("/")}
      >
        Employee Polls
      </Navbar.Brand>
      <Nav className="me-auto">
        <Nav.Link
          className={isActive("/") ? "active" : ""}
          onClick={() => navigate("/")}
        >
          Home
        </Nav.Link>
        <Nav.Link
          className={isActive("/leaderboard") ? "active" : ""}
          onClick={() => navigate("/leaderboard")}
        >
          Leaderboard
        </Nav.Link>
        {props.authedUser ? (
          <Nav.Link
            className={isActive("/new") ? "active" : ""}
            onClick={() => navigate("/add")}
          >
            New
          </Nav.Link>
        ) : null}
      </Nav>
      <Navbar.Collapse className="d-flex justify-content-end">
        <Image
          style={{
            width: "50px",
            height: "50px",
            marginRight: "10px",
          }}
          src={
            props.authedUser
              ? props.users[props.authedUser]?.avatarURL
              : "../../public/avatar/example.png"
          }
          alt="avatar"
          roundedCircle
        />
        <Navbar.Text data-testid="username-nav" style={{ marginLeft: "-10px" }}>
          {props.authedUser ? (
            "@" + props.authedUser
          ) : (
            <Navbar.Text
              style={{
                color: "white",
                marginLeft: "10px",
                marginRight: "10px",
                cursor: "pointer",
              }}
              onClick={() => {
                navigate("/login");
              }}
            >
              Login
            </Navbar.Text>
          )}
        </Navbar.Text>
        {props.authedUser ? (
          <Navbar.Text
            style={{
              color: "white",
              marginLeft: "10px",
              marginRight: "10px",
              cursor: "pointer",
            }}
            onClick={() => {
              localStorage.removeItem("authedUser");
              props.dispatch(setAuthedUser(""));
              navigate("/login");
            }}
          >
            Logout
          </Navbar.Text>
        ) : (
          ""
        )}
      </Navbar.Collapse>
    </Navbar>
  );
};

const mapStateToProps = ({ authedUser, users }) => {
  return {
    authedUser,
    users,
  };
};

export default connect(mapStateToProps)(Navigation);
