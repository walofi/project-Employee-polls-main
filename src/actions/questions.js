import { _saveQuestionAnswer, _saveQuestion } from "../../_DATA";
import { showLoading, hideLoading } from "react-redux-loading-bar";
import { addAnswer, addQuestionToUser } from "./users";

export const RECEIVE_QUESTIONS = "RECEIVE_QUESTIONS";
export const ANSWER_QUESTION = "ANSWER_QUESTION";
export const ADD_QUESTION = "ADD_QUESTION";

function addQuestion(question) {
  return {
    type: ADD_QUESTION,
    question,
  };
}

export function handleAddQuestion(question) {
  return (dispatch) => {
    dispatch(showLoading());

    return _saveQuestion(question)
      .then((question) => {
        dispatch(addQuestion(question));
        dispatch(
          addQuestionToUser({ qid: question.id, authedUser: question.author })
        );
      })
      .then(() => {
        dispatch(hideLoading());
        localStorage.removeItem("handleAnswer");
      });
  };
}

export function receiveQuestions(questions) {
  return {
    type: RECEIVE_QUESTIONS,
    questions,
  };
}

function answerQuestion({ authedUser, qid, answer }) {
  return {
    type: ANSWER_QUESTION,
    qid,
    authedUser,
    answer,
  };
}

export function handleAnswerQuestion(info) {
  return (dispatch) => {
    dispatch(showLoading());
    return _saveQuestionAnswer(info)
      .then(() => {
        dispatch(answerQuestion(info));
        dispatch(addAnswer(info));
      })
      .then(() => {
        dispatch(hideLoading());
        localStorage.removeItem("handleAnswer");
      })
      .catch((e) => {
        console.warn("Error in handleAnswerQuestion: ", e);
        alert("There was an error answering the question. Try again.");
      })
      .catch((e) => {
        console.error("Error in dispatching answerQuestion: ", e);
      });
  };
}
