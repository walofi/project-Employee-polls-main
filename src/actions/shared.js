import { _getUsers, _getQuestions } from "../../_DATA";
import { receiveUsers } from "./users";
import { receiveQuestions } from "./questions";
import { setAuthedUser } from "./authedUser";
import { showLoading, hideLoading } from "react-redux-loading-bar";

const AUTHED_ID = localStorage.getItem("authedUser")
  ? localStorage.getItem("authedUser")
  : "";

export function handleInitialData() {
  return (dispatch) => {
    dispatch(showLoading());
    _getUsers()
      .then((users) => {
        _getQuestions().then((questions) => {
          dispatch(receiveUsers(users));
          dispatch(receiveQuestions(questions));
          dispatch(setAuthedUser(AUTHED_ID));
          dispatch(hideLoading());
        });
      })
      .catch((error) => {
        alert(error);
      });
  };
}
