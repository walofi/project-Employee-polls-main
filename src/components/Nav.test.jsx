import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../store";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import Nav from "./Nav";
import { setAuthedUser } from "../actions/authedUser";

describe("Nav", () => {
  it("should render the component", () => {
    store.dispatch(setAuthedUser("br19"));

    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("should display username of logged in user", () => {
    store.dispatch(setAuthedUser("hb"));

    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );

    const usernameElement = component.getByTestId("username-nav");
    expect(usernameElement.textContent).toBe("@hb");
  });
});
