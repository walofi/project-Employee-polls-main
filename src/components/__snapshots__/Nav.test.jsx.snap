// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render the component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="container navbar navbar-expand navbar-dark bg-dark fixed-top"
      >
        <span
          class="navbar-brand"
          style="cursor: pointer;"
        >
          Employee Polls
        </span>
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="active nav-link"
            href="#"
            role="button"
            tabindex="0"
          >
            Home
          </a>
          <a
            class="nav-link"
            href="#"
            role="button"
            tabindex="0"
          >
            Leaderboard
          </a>
          <a
            class="nav-link"
            href="#"
            role="button"
            tabindex="0"
          >
            New
          </a>
        </div>
        <div
          class="d-flex justify-content-end navbar-collapse collapse"
        >
          <img
            alt="avatar"
            class="rounded-circle"
            style="width: 50px; height: 50px; margin-right: 10px;"
          />
          <span
            class="navbar-text"
            data-testid="username-nav"
            style="margin-left: -10px;"
          >
            @br19
          </span>
          <span
            class="navbar-text"
            style="color: white; margin-left: 10px; margin-right: 10px; cursor: pointer;"
          >
            Logout
          </span>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="container navbar navbar-expand navbar-dark bg-dark fixed-top"
    >
      <span
        class="navbar-brand"
        style="cursor: pointer;"
      >
        Employee Polls
      </span>
      <div
        class="me-auto navbar-nav"
      >
        <a
          class="active nav-link"
          href="#"
          role="button"
          tabindex="0"
        >
          Home
        </a>
        <a
          class="nav-link"
          href="#"
          role="button"
          tabindex="0"
        >
          Leaderboard
        </a>
        <a
          class="nav-link"
          href="#"
          role="button"
          tabindex="0"
        >
          New
        </a>
      </div>
      <div
        class="d-flex justify-content-end navbar-collapse collapse"
      >
        <img
          alt="avatar"
          class="rounded-circle"
          style="width: 50px; height: 50px; margin-right: 10px;"
        />
        <span
          class="navbar-text"
          data-testid="username-nav"
          style="margin-left: -10px;"
        >
          @br19
        </span>
        <span
          class="navbar-text"
          style="color: white; margin-left: 10px; margin-right: 10px; cursor: pointer;"
        >
          Logout
        </span>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
